readthedocs/doc_builder/environments.py

```diff
@@ -301,11 +301,35 @@ def run(self):
 
         self.start_time = datetime.utcnow()
         client = self.build_env.get_client()
+
+        # Create a copy of the environment to update PATH variable
+        environment = self._environment.copy()
+        # Default PATH variable
+        environment[
+            "PATH"
+        ] = "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+        # Add asdf extra paths
+        environment["PATH"] += (
+            f":/home/{settings.RTD_DOCKER_USER}/.asdf/shims"
+            f":/home/{settings.RTD_DOCKER_USER}/.asdf/bin"
+        )
+
+        if settings.RTD_DOCKER_COMPOSE:
+            environment["PATH"] += ":/root/.asdf/shims:/root/.asdf/bin"
+
+        # Prepend the BIN_PATH if it's defined
+        if self.bin_path:
+            path = environment.get("PATH")
+            bin_path = self._escape_command(self.bin_path)
+            environment["PATH"] = bin_path
+            if path:
+                environment["PATH"] += f":{path}"
+
```

Comment:
This logic feels really weird to me. We're overwriting then appending the original path if it exists, but the variable names don't make that clear. I think we want something closer to:

```suggestion
        if self.bin_path:
            original_path = environment.get("PATH")
            # SECURITY CRITICAL: Escape user input
            escaped_bin_path = self._escape_command(self.bin_path)
            environment["PATH"] =  f"{escaped_bin_path}:{original_path}"
```

Reply:
We need the `if` because `original_path` could be `None`.

I'm taking the suggestion of the variable names, tho.